pkg/config/config.go

```diff
@@ -162,3 +152,67 @@ func ExportKubeProxyConfig(fullState *cluster.FullState, dataDir string) error {
 
 	return nil
 }
+
+func ConfigurePrivateRegistries(ctx context.Context, fullState *cluster.FullState, registriesTLS []string) error {
+	privateRegistryConfig := fullState.CurrentState.RancherKubernetesEngineConfig.PrivateRegistries
+	if len(privateRegistryConfig) <= 0 {
+		return nil
+	}
+	if _, err := os.Stat(privateRegistryPath); err != nil {
+		if os.IsNotExist(err) {
+			r := registries.Registry{}
+			r.Configs = make(map[string]registries.RegistryConfig)
+			for _, reg := range privateRegistryConfig {
+				endpoint := reg.URL
+				if endpoint == "" {
+					endpoint = "docker.io"
+				}
+				if reg.User != "" && reg.Password != "" {
+					r.Configs[endpoint] = registries.RegistryConfig{
+						Auth: &registries.AuthConfig{
+							Username: reg.User,
+							Password: reg.Password,
+						},
+						TLS: getRegistryTLSConfig(endpoint, registriesTLS),
+					}
+				}
+			}
+
+			regBytes, err := yaml.Marshal(r)
+			if err != nil {
+				return err
+			}
+			if err := ioutil.WriteFile(privateRegistryPath, regBytes, 0600); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func getRegistryTLSConfig(endpoint string, registriesTLS []string) *registries.TLSConfig {
+	if len(registriesTLS) <= 0 {
+		return nil
+	}
+	var caCert, cert, key, url string
+	for _, registryTLS := range registriesTLS {
```

Comment:
Can we safely assume that the URLs passed in here are all valid? If not, should we first validate them before splitting them into parts?